src/viking.js

```diff
@@ -1,11 +1,119 @@
-// Soldier
-function Soldier() {}
+
+function Soldier (healthArg, strengthArg) {
+	this.health = healthArg;
+	this.strength = strengthArg;
+	//this.name = nameArg; // I would pass 3 arguments to Soldier for the subclasses to inherit.
+}
+
+Soldier.prototype.attack = function () {
+	return this.strength;
+}
+
+Soldier.prototype.receiveDamage = function (damage) {
+	if (damage !== 0) {
+		this.health -= damage;
+	}
+
+}
+
+//console.log('Soldier methods:', Soldier.prototype);
+	
 
 // Viking
-function Viking() {}
+function Viking (nameArg, healthArg, strengthArg) {
+	this.name = nameArg;
+	Soldier.call(this, healthArg, strengthArg);
+}
+
+Viking.prototype = Object.create(Soldier.prototype); // I am not sure if it will overwrite revieceDamage function
+Viking.prototype.receiveDamage = function (damage) {
+	this.health -= damage;
+	if(this.health > 0) {
+		return `${this.name} has received ${damage} points of damage`;
+	} else if (this.health <= 0) {
+		return `${this.name} has died in act of combat`;
+	}
+}
+Viking.prototype.battleCry = function () {
+	return 'Odin owns you all!!'
+}
+
 
 // Saxon
-function Saxon() {}
+function Saxon(healthArg, strengthArg) {
+	this.name = 'Saxon';
+	Soldier.call(this, healthArg, strengthArg)
+}
 
+Saxon.prototype = Object.create(Soldier.prototype);
+Saxon.prototype.receiveDamage = function (damage) {
+	this.health -= damage
+	if (this.health < 0) {
+		return `A ${this.name} has received ${damage} points of damage`;
+	} else if (this.health <= 0) {
+		return `A ${this.name} has died in combat`;
+	}
+}
 // War
-function War() {}
+function War() {
+	this.vikingArmy = [];
+	this.saxonArmy = [];
+}
+
+War.prototype.addViking = function (Viking) { // I dont understand if viking its a parameter why I am asked to pass Viking object, i think it should work with any argument to whom i will later pass Viking obj
+	this.vikingArmy.push(Viking); // I wonder if there is a method to push several objects to an array or if it would haev been possible to do it with a loop
+	// this.vikingArmy.push(Viking); 
```

Comment:
The idea here is to create a function so jasmine can call it by passing a viking object so it gets pushed into the vikingArmy array, it's only necessary to do one push.